internal/engine/uiresource/subscriber.go

```diff
@@ -62,13 +72,38 @@ func (s *Subscriber) OnChange(ctx context.Context, st store.RStore, summary stor
 			// If there's a current version but no last version of this resource,
 			// create it.
 			st.Dispatch(NewUIResourceCreateAction(current))
+			err := s.client.Create(ctx, current.DeepCopy())
+			if err != nil {
+				st.Dispatch(store.NewErrorAction(fmt.Errorf("creating resource %s: %v", current.Name, err)))
+				return
+			}
 			s.lastResources[name] = current
 			continue
 		}
 
 		if !equality.Semantic.DeepEqual(last.Status, current.Status) {
 			// If the current version is different than the last version, update it.
 			st.Dispatch(NewUIResourceUpdateStatusAction(current))
+
+			freshResource := v1alpha1.UIResource{}
+			err := s.client.Get(ctx, types.NamespacedName{Name: current.Name}, &freshResource)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					delete(s.lastResources, name)
```

Comment:
I feel like this could cause trouble unless with the latest fixes cache mean we won't **ever** return not found for a new/existing object - if that happens, this will result in a create next invocation and get an already exists error (which will throw a store error).

That said, the alternative (ignoring, hoping it's stale cache and moving on) caused a whole different set of problems for KD when cache never caught up, so maybe we really do need to start doing some sort of upsert.

FWIW I just did some quick and dirty testing and am NOT seeing not founds anymore on `master` even with a bunch of entities, so this is probably fine as is.